### PR TITLE
DM-39747: Use correct Python versioning for PyPI

### PR DESCRIPTION
**Pull request by arunkannawadi**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

The above checklist is not applicable for this PR.

### CODE REVIEW

**Comment by timj** [comment]
That explains why we haven't had a PyPI release since March. Broken in #93. Thanks.

File: COPYRIGHT
Comment (timj) [inline COPYRIGHT:6]: All copyrights were transferred from LSST Corp to AURA so this is not the right fix and the original AURA line above was the right one.

Comment (arunkannawadi) [reply COPYRIGHT:6]: So just the last line needs to be removed, right?

Comment (timj) [reply COPYRIGHT:6]: The original AURA dates 2008-2014 were fine and this line can be removed.

File: COPYRIGHT
Comment (timj) [inline COPYRIGHT:5]: As far as I can tell there is one commit from Tokyo and is 2 lines of code. Generally copyright assignment requires more of a contribution than that and that is why I did not add this line originally (I had already gone through the code to generate the original copyright file contents around 2020). Please remove.

**Review by timj — APPROVED** [review]
Thanks for fixing the PyPI uploads. Unfortunately a failure in the PyPI action doesn't trigger an email to anyone.

**Comment by arunkannawadi** [comment]
@timj I'll merge after you take one final look at the COPYRIGHT. Since this is used outside of LSST, I wanted to make sure that it's accurate.

File: COPYRIGHT
Comment (timj) [inline COPYRIGHT:1]: Please add back the more modern dates for AURA. I think you went back to just the old LSST Corp dates.

I think it should be:

```suggestion
Copyright 2008-2014, 2016, 2017, 2019, 2020, 2022-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
```

**Review by timj — APPROVED** [review]

File: README.rst
Comment (timj) [inline README.rst:7]: I don't understand what this gives me? This should always be reporting test passing because we aren't allowed to merge to main without it. Was this showing failed on main because of the PyPI upload failure for any time where main and a tag coincided?

File: COPYRIGHT
Comment (arunkannawadi) [reply COPYRIGHT:1]: Oh, I changed it locally and forgot to push

File: README.rst
Comment (arunkannawadi) [reply README.rst:7]: No, the last step doesn't get run when you merge to main, but only on tag. The status of this badge would have turned to failed (in red). Similar to how the red X shows up next to the most recent commit on `main`.

This is the workflow log that it would report: https://github.com/lsst/pex_config/actions/runs/5276812239/jobs/9544053729

Comment (timj) [reply README.rst:7]: I'm still confused because that link at the moment shows "build_and_test: passing" -- it does not show the failure that I can see if I go to the main GitHub page and click on the little red "x" against current main commit.

Comment (arunkannawadi) [reply README.rst:7]: It shows green now because the most recent run of this workflow passed (and didn't try uploading to PyPI). If I merged just the badge tonight and hold off the main fix until tomorrow, I believe it would turn red by then.

File: COPYRIGHT
Comment (arunkannawadi) [reply COPYRIGHT]: Hmm, the `copyright.py` script reports an activity from AURA in 2018 as well. Here's one: https://github.com/lsst/pex_config/pull/28

File: README.rst
Comment (timj) [reply README.rst:7]: If it always is showing the most recent build from a PR unrelated to main I don't think we want this. It doesn't tell people anything actionable.

File: COPYRIGHT
Comment (timj) [reply COPYRIGHT]: I decided not to include it because it was a trivial change that should not really result in asserting a copyright claim.

Comment (timj) [reply COPYRIGHT]: Also, if there is a copyright.py script that people are using it needs to be fixed to understand that LSST Corp no longer owns copyright.

File: README.rst
Comment (arunkannawadi) [reply README.rst]: Fair enough, removed.

File: COPYRIGHT
Comment (arunkannawadi) [reply COPYRIGHT]: This is what I see in the dev guide and it explicitly returns LSST Corporation, despite the docstring mentioning the transfer: https://github.com/lsst-dm/dm_dev_guide/blob/434dec6845dc2ae28eeb9f7c9b918617aff8eee1/legal/copyright.py#L115

Comment (timj) [reply COPYRIGHT]: Okay. It's wrong.

**Pull request merged**
